WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs

```diff
@@ -121,18 +121,27 @@ public object GetStatus()
 		}
 
 		[JsonRpcMethod("build")]
-		public string BuildTransaction(PaymentInfo[] payments, OutPoint[] coins, int feeTarget, string password = null)
+		public string BuildTransaction(PaymentInfo[] payments, OutPoint[] coins, int feeTarget = 0, decimal feeRate = 0, string password = null)
 		{
 			Guard.NotNull(nameof(payments), payments);
 			Guard.NotNull(nameof(coins), coins);
-			Guard.InRangeAndNotNull(nameof(feeTarget), feeTarget, 2, Constants.SevenDaysConfirmationTarget);
 			password = Guard.Correct(password);
+			
+			var argFeeRate = new FeeRate(feeRate);
+
+			static bool InRange<T>(IComparable<T> val, T min, T max) =>
+				val.CompareTo(min) >= 0 && val.CompareTo(max) <= 0;
+
+			var feeStrategy = (feeRate, feeTarget) switch
+			{
+				(_, 0) when InRange(argFeeRate, Constants.MinRelayFeeRate, Constants.AbsurdlyHighFeeRate) => FeeStrategy.CreateFromFeeRate(argFeeRate),
```

Comment:
The default value is zero, it is a luck that these values cannot be zero - can we make this more reliable to detect if those argumenta are not filled? Probably having a nullable argument, so if it missing the arg will be null and not zero. @lontivero how are you doing these kind of thing in RPC? Nullable had an effect on API side in the backend, probably here too, can you elaborate on this and make a standard?

Reply:
The values cannot be zero but that is not a luck, the values were explicitly set to zero because it doesn't matter whether the value zero is or not passed it is invalid anyway. I will change it but adding a null instead of a zero will not make it more reliable.

Nulleable types can have a side effect and i think the only way to be sure is by testing. However, here we have full control because the rpc server is the one parsing the json request and feeding the methods. Anyway, I will double check.

Reply:
Whatever you decide is OK for me.